CMakeLists.txt

```diff
@@ -258,6 +258,13 @@ add_executable(savegamedump WIN32
 	src/PngWriter.cpp
 )
 
+add_custom_target(codedoc ALL)
+add_custom_command(TARGET codedoc POST_BUILD
+	COMMENT "Building naturaldoc for LUA API"
+	COMMAND bash -c "mkdir -p codedoc"
+	COMMAND bash -c "naturaldocs -i src/ -i data/libs/ -xi src/data/ -o HTML codedoc/ -p nd/ -do -ro -s Default Local"
+	WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
+
```

Comment:
Thoroughly untested, but you might want to simply add a check for naturaldocs :
```cmake
find_program(NATURALDOCS NAMES naturaldocs)
if (NATURALDOCS)
add_custom_target(...)
endif()
```

Reply:
Added. "works on my system". Thanks.

(Still interested in how to do a `make codedoc`)